sdk/core/azure-core/src/main/java/com/azure/core/util/QueryParameter.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a single parameter to be added to a query string.
+ *<p>
+ * If multiple values are added to a query string with the same name (case-insensitive), then the values will be
+ * appended at the end of the same {@link QueryParameter} with commas separating them.
+ */
+public final class QueryParameter {
+    private final String name;
+
+    // this is the actual internal representation of all values
+    private final List<String> values;
+
+    // but we also cache it to faster serve our public API
+    private volatile String cachedStringValue;
+
+    /**
+     * Create a QueryParameter instance using the provided name and value.
+     *
+     * @param name the name of the parameter.
+     * @param value the value of the parameter.
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public QueryParameter(String name, String value) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+        this.name = name;
+        this.values = new LinkedList<>();
+        this.values.add(value);
```

Comment:
I would consider this to be the common case, in which case I would suggest lazily instantiating the `values` list only when it becomes necessary to have a second value.

Reply:
Done.